### PR TITLE
Pull in latest CoreCLR, CoreFX, and CoreFX Test Binaries, Generate Stress Suite And Submit to Helix

### PR DESCRIPTION
**Pull request by bryanar**
WIP

### CODE REVIEW

File: src/stress.codegen/TestAssemblyLoader.cs [hunk 1]
```diff
@@ -111,21 +112,28 @@ private static string FindReferenceAssemblyInPaths(string assmName, IEnumerable<
...
 
-                return discoverer.GetTests(_assembly);
+                var tests = discoverer.GetTests(_assembly);
+
+                if (tests.Length > 0)
+                {
+                    CodeGenOutput.Info($"{_assembly.Assembly.FullName}: {tests.Length} tests discovered from assembly");
+                }
+
+                return JsonConvert.SerializeObject(tests);
```
Comment (bryanar) [inline src/stress.codegen/TestAssemblyLoader.cs]: Serializing in to JSON allows to us to remove the crutch where we can't use certain types because the runtime doesn't know how to serialize/marshall it. So here we say "Okay CLR serializier, just marshall this string (wink) and we'll be happy."

File: src/stress.codegen/UnitTestInfo.cs [hunk 2]
```diff
@@ -31,7 +31,6 @@ public ProjectJsonDependencyInfo PackageInfo
     }
 
 
-    [Serializable]
```
Comment (bryanar) [inline src/stress.codegen/UnitTestInfo.cs]: No need for these now.

File: src/stress.codegen/UnitTestSelector.cs [hunk 3]
```diff
@@ -131,19 +122,7 @@ public static UnitTestInfo[] ReadCacheFromFile(string path)
...
-            {
-                using (StreamReader reader = new StreamReader(fs))
-                {
-                    JsonTextReader jReader = new JsonTextReader(reader);
-
-                    // Call the Deserialize method to restore the object's state.
-                    cache = serializer.Deserialize<UnitTestInfo[]>(jReader);
-                }
-            }
+            cache = JsonConvert.DeserializeObject<UnitTestInfo[]>(File.ReadAllText(path));
```
Comment (bryanar) [inline src/stress.codegen/UnitTestSelector.cs]: Cleaning up/simplifying

File: test/genstress.proj [hunk 4]
```diff
@@ -90,17 +88,32 @@
...
+        <ZipFileExtractToDirectory Condition="!Exists('%(RootDir)%(Directory)%(ZipFile.FileName)')"
+            SourceArchive="%(ZipFile.FullPath)"
+            DestinationDirectory="$(UnitTestExtractDir)\%(filename)%(extension)"
+            OverwriteDestination="true" />
+        
+        <Copy SourceFiles="@(JsonFile)" DestinationFiles="@(JsonFile->Replace('$(UnitTestDirectory)','$(UnitTestExtractDir)'))" />
+    </Target>
+
+    <Target Name="SelectTestZips">
+        <Error Condition="!Exists('$(MSBuildThisFileDirectory)\..\AnyOS.AnyCPU.Debug-Test-Archive-List.txt')" Text="Can't find AnyOS Test List.'" />
```
Comment (bryanar) [inline test/genstress.proj]: This is placed here temporarily. Waiting on CoreFX tests to come back online to complete this piece.

File: src/stress/StressEssentials.ps1 [hunk 5]
```diff
@@ -1,61 +1,227 @@
-﻿$workingDirectory=$env:TEMP
-$productArchitecture="TBD - passed in by VSTS likely."
-$productConfiguration="chk/ret: TBD - passed in by VSTS likely."
+# Ideally, we should use the current logged in user but this appears to require domain joining. 
 
-$productDirectory="$workingDirectory/product/$productArchitecture/$productConfiguration"
-$testsDirectory="$workingDirectory/tests/"
+param([String]$DropPat, [String]$WorkingDirectory, [Switch]$DebuggingBinaries = $false)
```
Comment (bryanar) [inline src/stress/StressEssentials.ps1]: Added a switch to the stress essentials. When invoking the stress essentials, if you pass in /DebuggingBinaries it will retrieve chk build of CoreCLR and debug builds of CoreFX.

File: test/suiteconfig/16hr.suite.json [hunk 6]
```diff
@@ -1,5 +1,5 @@
 {
-  "Host": "./corerun",
```
Comment (bryanar) [inline test/suiteconfig/16hr.suite.json]: Note: Going forward, be careful here. This string exactly ends up in our executing batch and shell scripts that invoke our tests. Using ./corerun doesn't fly in batch. In the positive side, we will prepend the ./ when producing the shell file, so that we are using the host that sits in the test directory.

File: build.cmd [hunk 7]
```diff
@@ -41,7 +41,7 @@ call :build %*
 goto :AfterBuild
 
 :build
-%_buildprefix% msbuild "%_buildproj%" /nologo /maxcpucount /verbosity:minimal /nodeReuse:false /fileloggerparameters:Verbosity=normal;LogFile="%_buildlog%";Append %* %_buildpostfix%
+%_buildprefix% msbuild "%_buildproj%" /nologo /verbosity:minimal /nodeReuse:false /fileloggerparameters:Verbosity=diagnostic;LogFile="%_buildlog%";Append %* %_buildpostfix%
```
Comment (schaabs) [inline build.cmd]: need to revert the /maxcpucount and logfile verbosity back to normal

File: src/stress.codegen/LoadSuiteConfig.cs [hunk 8]
```diff
@@ -26,35 +26,14 @@ public LoadSuiteConfig()
...
-            {
-                using (StreamReader reader = new StreamReader(fs))
-                {
-                    JsonTextReader jReader = new JsonTextReader(reader);
-
-                    // Call the Deserialize method to restore the object's state.
-                    config = serializer.Deserialize<LoadSuiteConfig>(jReader);
-                }
-            }
+            config = JsonConvert.DeserializeObject<LoadSuiteConfig>(File.ReadAllText(path));
```
Comment (schaabs) [inline src/stress.codegen/LoadSuiteConfig.cs]: > config = JsonConvert.DeserializeObject<LoadSuiteConfig>(File.ReadAllText(path)); [](start = 11, length = 81)

Is there a specific reason for these changes?  The general functionality is the same, but it requires loading the entire config string into memory, and there are differences in how the file written.

Comment (bryanar) [inline src/stress.codegen/LoadSuiteConfig.cs]: Improves readability - same functionality. I don't think that loading the entire config file in to memory is going to hurt anything given that it happens once. Also, what's different in the way that the file gets written?

Comment (schaabs) [inline src/stress.codegen/LoadSuiteConfig.cs]: The Serlialize method above originally failed if the specified path was a preexisting file.  The change updates it to overwrite.

Comment (bryanar) [inline src/stress.codegen/LoadSuiteConfig.cs]: Do we want this?

Comment (schaabs) [inline src/stress.codegen/LoadSuiteConfig.cs]: I think we should revert the changes to this file since they're not necessary for the updates your making in the pull request.

Comment (bryanar) [inline src/stress.codegen/LoadSuiteConfig.cs]: Agreed

**Comment by schaabs** [comment]
:shipit:
